RELEASE_PROCESS.md

```diff
@@ -20,92 +40,296 @@ Regular releases concern only major and minor releases. We do not decide up fron
+
+This issue should be updated as each of the following steps are completed.
+
+A [template for a new release issue](https://github.com/asyncapi/spec/blob/master/.github/ISSUE_TEMPLATE/release.md) is available.
+
+An example is the [release issue for the 2.3.0](https://github.com/asyncapi/spec/issues/675) release.
+
+### Step 3 - create release branches
+
+Release branches must have a year and a month of the release as prefix: {YEAR_OF_RELEASE}-{MONTH_OF_RELEASE}-release.
+
+For example, a release created in September 2021 has a `2021-09-release` release branch.
+
+<img src="./assets/release_process/create_branch.png" alt="This image shows part of the GitHub UI that shows how you can create a new branch using default branch as a base." height="300">
 
-At the beginning of the release cycle, we need to have a new release branch created in some repositories. The reason is that [contribution guide](CONTRIBUTING.md) for the specification requires changes in different projects to accept a proposal. The following repositories are involved:
+At the beginning of the release cycle, we need to have a new release branch created in the following repositories:
 - [spec](https://github.com/asyncapi/spec) where contributor works with the specification file,
 - [spec-json-schemas](https://github.com/asyncapi/spec-json-schemas) where contributor pushes changes to JSON Schema of the spec,
 - [parser-js](https://github.com/asyncapi/parser-js) where contributor makes necessary changes in the JavaScript Parser.
 
-Release branch must have a year and a month of the release as prefix: {YEAR_OF_RELEASE}-{MONTH_OF_RELEASE}-release. For example, a release created in September 2021 has a `2021-09-release` release branch.
+The [release coordinator](#%22release-coordinator%22) should decide what the branch name needs to be, and contact the [code owners](#code-owners) for each repository to ask them to create the branches for them.
 
-<img src="./assets/release_process/create_branch.png" alt="This image shows part of the GitHub UI that shows how you can create a new branch using default branch as a base." height="300">
 
-Once feature branches are created, there must be some initial configuration done in each repository.
+### Step 4 - update release branches
+
+Once [release branches are created](#step-3---create-release-branches), there are some initial changes that need to be made.
+
+The process for doing this is the same for each of these:
+- the [release coordinator](#%22release-coordinator%22) should create a fork of the relevant repository for these changes
+- the commit message for the change should start with `chore:`
+- the change should be contributed in a pull request targeting the [release branch](#step-3---create-release-branches)
+- the [release coordinator](#%22release-coordinator%22) will need to ask the [code owners](#code-owners) for the relevant repository to approve and merge this pull request
+
+#### Update version numbers in official examples
```

Comment:
Nitpick: Could we prefix the subtitles with a step number, such as `4.1`, `4.2` etc? I'm following this guide right now and I missed that info which would make it easier for me to visualize the steps to follow.

Reply:
good idea 👍